lib/command_runner.dart

```diff
@@ -158,13 +180,32 @@ class CommandRunner<T> {
 
           command.usageException('Missing subcommand for "$commandString".');
         } else {
+          var requested = argResults.rest[0];
+
+          // Build up a help message containing similar commands, if found.
+          var similar = StringBuffer();
+          if (suggestedCommandsMaxEditDistance > 0) {
+            var possible = similarCommands(requested);
+            if (possible.isNotEmpty) {
+              similar.write(' The most similar command');
+              if (possible.length == 1) {
+                similar.writeln(' is:');
+              } else {
+                similar.writeln('s are:');
+              }
+              for (var command in possible) {
+                similar.writeln('  ${command.name}');
+              }
+            }
+          }
```

Comment:
Also, I would move all this into the aforementioned method. Just have a single method that gives you the suggestion text and keep this one relatively simple.